mbed_host_tests/host_tests_conn_proxy/conn_primitive_serial.py

```diff
@@ -84,6 +87,9 @@ def reset_dev_via_serial(self, delay=1):
 
     def read(self, count):
         """! Read data from serial port RX buffer """
+        # TIMEOUT: Since read is called in a loop, wait for self.timeout period before calling serial.read(). See
+        # comment on serial.Serial() call above about timeout.
+        sleep(self.timeout)
```

Comment:
@sg- buffering of data is done here https://github.com/mazimkhan/htrun/blob/80d13359cef7485bb2b2bc01fd6c9cdc332e5244/mbed_host_tests/host_tests_conn_proxy/conn_proxy.py#L36
For pyserial ```timeout=0``` means non blocking i.e. ```read()``` returns immediately. For non-blocking it has to be set as ```None```. 
Based on the baud rate (max 115200) the data speed is limited and the timeout is short enough. Reducing timeout means aggressive polling for data. Since the data rate is already limited by the baud rate we don't need aggressive polling. But yes shorter timeout may help in preciseness in timing tests.